### PR TITLE
Map update

### PR DESCRIPTION
**Pull request by kgao**
Connect API callback cards with map update (refresh)

### CODE REVIEW

File: clientapp/src/components/Map.jsx [hunk 1]
```diff
@@ -1,74 +1,179 @@
...
+        this.state = {mapInfo: this.props.mapInfo, cardInfo: []};       
     }
 
     componentDidMount(){
         window.onload = this.loadMapScenario;
     }
 
+    componentDidUpdate(){
+      //this.setState({cardInfo:() => this.props.cardInfo});
+      this.state.cardInfo = this.props.cardInfo;
```
Comment (davidmjiang) [inline clientapp/src/components/Map.jsx:21]: state is typically used when something internal to the component may cause it to change. In this case, we are changing based on input from parent component, so it seems like we can just read directly from props, without reading from state. So componentDidUpdate() could look like:

`this.loadMapScenario()` and then in loadMapScenario, read from this.props.cardInfo

File: clientapp/src/components/Map.jsx [hunk 2]
```diff
@@ -1,74 +1,179 @@
...
-                endDate: "",
-                description: "",
-                lat: this.props.mapInfo.lat,
-                lon: this.props.mapInfo.lon,
-                address: this.props.mapInfo.address
-            }
-        ]
-        let map = BingMap.createMap();
+        const cards = this.state.cardInfo;
+        //TODO: load list of card
```
Comment (davidmjiang) [inline clientapp/src/components/Map.jsx:41]: remove?

Comment (kgao) [reply clientapp/src/components/Map.jsx]: done

File: clientapp/src/components/Map.jsx (hunk 1 above)
Comment (kgao) [reply clientapp/src/components/Map.jsx]: sure

**Pull request merged**